src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,712 @@
+                    incompatible_config, "BOOST_WAVE_PRAGMA_KEYWORD",
+                    get_main_pos());
+                return;
+            }
+
+            // BOOST_PP_STRINGIZE((BOOST_WAVE_STRINGTYPE))
+            ar& make_nvp("string_type", string_type_str);
+            if (string_type_str != BOOST_PP_STRINGIZE((BOOST_WAVE_STRINGTYPE))) {
+                BOOST_WAVE_THROW_CTX((*this), preprocess_exception,
+                    incompatible_config, "BOOST_WAVE_STRINGTYPE", get_main_pos());
+                return;
+            }
+
+            try {
+                // read in the useful bits
+                ar& make_nvp("language_options", language);
+                ar& make_nvp("macro_definitions", macros);
+                ar& make_nvp("include_settings", includes);
+            }
+            catch (boost::wave::preprocess_exception const& e) {
+                // catch version mismatch exceptions and call error handler
+                get_hooks().throw_exception(derived(), e);
+            }
+        }
+        BOOST_SERIALIZATION_SPLIT_MEMBER()
+#endif
+
+    private:
+        // the main input stream
+        target_iterator_type first;         // underlying input stream
+        target_iterator_type last;
+        std::string filename;               // associated main filename
+        bool has_been_initialized;          // set cwd once
+#if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
+        std::string current_filename;       // real name of current preprocessed file
+#endif
+        std::string current_relative_filename;        // real relative name of current preprocessed file
+
+        boost::wave::util::if_block_stack ifblocks;   // conditional compilation contexts
+        include_paths includes;    // lists of include directories to search
+        iteration_context_stack_type iter_ctxs;       // iteration contexts
+        macromap_type macros;                         // map of defined macros
+        boost::wave::language_support language;       // supported language/extensions
+        hook_policy_type hooks;                       // hook policy instance
+    };
+
+}
+
+
+namespace nbl::asset::impl
 {
-    struct DXC {
+    struct DXC 
+    {
         ComPtr<IDxcUtils> m_dxcUtils;
         ComPtr<IDxcCompiler3> m_dxcCompiler;
     };
+
+
+    // for including builtins 
+    struct load_file_or_builtin_to_string
+    {
+        template <typename IterContextT>
+        class inner
+        {
+        public:
+            template <typename PositionT>
+            static void init_iterators(IterContextT& iter_ctx, PositionT const& act_pos, boost::wave::language_support language)
+            {
+                using iterator_type = typename IterContextT::iterator_type;
+
+                std::string filepath(iter_ctx.filename.begin(), iter_ctx.filename.end());
+                auto inclFinder = iter_ctx.ctx.get_hooks().m_includeFinder;
+                if (inclFinder) 
```

Comment:
As I've already stated multiple times:
- https://github.com/Devsh-Graphics-Programming/Nabla/pull/608#discussion_r1399347272
- https://github.com/Devsh-Graphics-Programming/Nabla/pull/608#discussion_r1401155570
**Just assert the include finder is not null**